### PR TITLE
feat: Adds auto-layout method for laying out blocks in a plan

### PR DESCRIPTION
**Pull request by hofmeister**

### CODE REVIEW

File: src/planner/auto-layout.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    nodeWidth: number;
+    marginX: number;
+    marginY: number;
+}
+
+function toEdgeId(providerId: string, consumerId: string) {
+    return [providerId, consumerId].sort().join('-');
+}
+
+export function applyAutoLayout(
```
Comment (wejendorp) [inline src/planner/auto-layout.ts:27]: Can you add a description of the general idea of the layout logic? Will probably make reading the code a lot easier for me.

Comment (hofmeister) [reply src/planner/auto-layout.ts:38]: Done

Comment (wejendorp) [reply src/planner/auto-layout.ts:38]: On it ✏️

File: src/planner/auto-layout.ts [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        blockSizes[blockId] = calculateBlockSize({
+            nodeSize: opts.nodeSize,
+            blockType: BlockTypeProvider.get(blockDef.kind),
+            blockDefinition: blockDef,
+            blockMode: BlockMode.SHOW,
+        });
+    });
+
+    let columnIndex = 0;
+    let currentX = CANVAS_OFFSET_LEFT;
```
[the remark this answers is not in the document]
Comment (wejendorp) [reply src/planner/auto-layout.ts:135]: (And similar Q for the Y offset)

[the remark this answers is not in the document]
Comment (hofmeister) [reply src/planner/auto-layout.ts:135]: We've got some margin things in the plan editor itself that requires this

**Pull request merged**